packages/2d/src/components/SVG.ts

```diff
@@ -0,0 +1,387 @@
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+import {BBox, SerializedVector2, Vector2} from '@motion-canvas/core/lib/types';
+import {computed, signal} from '../decorators';
+import {Shape, ShapeProps} from './Shape';
+import {Node} from './Node';
+import {DesiredLength, PossibleCanvasStyle} from '../partials';
+import {useLogger} from '@motion-canvas/core/lib/utils';
+import {Path} from './Path';
+import {Rect} from './Rect';
+import {
+  clampRemap,
+  easeInOutSine,
+  tween,
+} from '@motion-canvas/core/lib/tweening';
+import {Layout} from './Layout';
+import diffSequence from 'diff-sequences';
+
+interface ParsedSVG {
+  size: Vector2;
+  nodes: Array<Node>;
+}
+
+interface SVGDiff {
+  fromSize: Vector2;
+  toSize: Vector2;
+  inserted: Array<Node>;
+  deleted: Array<Node>;
+  transformed: Array<{
+    from: Node;
+    to: Node;
+  }>;
+}
+
+export interface SVGProps extends ShapeProps {
+  svg?: SignalValue<string>;
+}
+
+export class SVG extends Shape {
+  @signal()
+  public declare readonly svg: SimpleSignal<string, this>;
+  public wrapper: Node;
+
+  public constructor(props: SVGProps) {
+    super(props);
+    this.wrapper = new Node({});
+    this.wrapper.children(this.parsedNodes);
+    this.add(this.wrapper);
+  }
+
+  protected override desiredSize(): SerializedVector2<DesiredLength> {
+    const custom = super.desiredSize();
+    const {x, y} = this.parsed().size.mul(this.wrapper.scale());
+    return {
+      x: custom.x ?? x,
+      y: custom.y ?? y,
+    };
+  }
+
+  @computed()
+  private parsed() {
+    return this.parseSVG(this.svg());
+  }
+
+  @computed()
+  private parsedNodes() {
+    return this.parsed().nodes;
+  }
+
+  private parseSVG(svg: string): ParsedSVG {
+    const container = document.createElement('div');
+    container.innerHTML = svg;
+
+    const svgRoot = container.querySelector('svg')!;
+    const {x, y, width, height} = svgRoot.viewBox.baseVal;
+    const viewBox = new BBox(x, y, width, height);
+    const size = new Vector2(
+      svgRoot.width.baseVal.value,
+      svgRoot.height.baseVal.value,
+    );
+    const scale = size.div(viewBox.size);
+    const center = viewBox.center;
+
+    const rootTransform = new DOMMatrix()
+      .scale(scale.x, scale.y)
+      .translate(-center.x, -center.y);
```

Comment:
Could be:
```suggestion
      .scaleSelf(scale.x, scale.y)
      .translateSelf(-center.x, -center.y);
```
to not create multiple instances of the matrix.